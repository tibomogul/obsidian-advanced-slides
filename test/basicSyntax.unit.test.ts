import { ObsidianMarkdownPreprocessor } from 'src/obsidianMarkdownPreprocessor';
import { when } from 'ts-mockito';
import { MockedObsidianUtils, obsidianUtils as utilsInstance } from './__mocks__/mockObsidianUtils';
import { prepare } from './testUtils';

test('Basic Markdown Syntax > Headers', () => {
	const input = `# This is a heading 1
## This is a heading 2
### This is a heading 3 
#### This is a heading 4
##### This is a heading 5
###### This is a heading 6`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Text style', () => {
	const input = `*This text will be italic*

_This will also be italic_

**This text will be bold**

__This will also be bold__

%%This is a Comment(Can't see it)%%

_You **can** combine them_

---

Any word wrapped with two tildes (like ~~this~~) will appear crossed out.

Any word wrapped with two equal signs (like ==this==) will appear as highlighted.`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Lists', () => {
	const input = `- Item 1
- Item 2
	- Item 2a
	- Item 2b

---

1. Item 1
1. Item 2
1. Item 3
   1. Item 3a
   1. Item 3b
`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Images', () => {
	//TODO: bugfix getAbsolutePath
	when(MockedObsidianUtils.getAbsolutePath('Image.jpg.md')).thenCall(arg => {
		return null;
	});

	when(MockedObsidianUtils.getAbsolutePath('Image.jpg|100.md')).thenCall(arg => {
		return null;
	});

	when(MockedObsidianUtils.getAbsolutePath('Image.jpg|300x100 <!-- element style="object-fit: cover" -->.md')).thenCall(arg => {
		return null;
	});

	when(MockedObsidianUtils.findFile('Image.jpg')).thenCall(arg => {
		return '/documentation/Image.jpg';
	});

	const input = `
Insert image with standard markdown syntax

![Image](https://picsum.photos/id/1006/500/300)

---

Insert image that lies outside of your vault

![Image](file:///Users/testUser/Desktop/howToUse.png)

---

Insert image with obsidian markdown syntax

![[Image.jpg]]

---

Scale image to a width of 100 px

![[Image.jpg|100]]

---

Scale image to a width of 300x100 px

![[Image.jpg|300x100]] <!-- element style="object-fit: cover" -->

`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Links', () => {
	const input = `External Links

http://obsidian.md - automatic!

[Obsidian](http://obsidian.md)

---

Obsidian URI links

[Link to note](obsidian://open?path=D:%2Fpath%2Fto%2Ffile.md)

[Link to note](obsidian://open?vault=MainVault&file=MyNote.md)

---

Internal Links

This [[Internal link]] will be displayed as normal text

This [[Internal link|Link]] will use its alias for displaying

`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Links', () => {
	const input = `---
enableLinks: true
---

Internal Links

This [[Internal link]] will be displayed as normal text

This [[Internal link|Link]] will use its alias for displaying

`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Embeds', () => {

	when(MockedObsidianUtils.parseFile('Obsidian.md', 'Link')).thenCall(arg => {
		return 'Link to Obsidian Homepage: http://obsidian.md';
	});

	when(MockedObsidianUtils.parseFile('Obsidian.md', null)).thenCall(arg => {
		return `# Notes about Obsidian
A knowledge base that works on local Markdown files

# Link
Link to Obsidian Homepage: http://obsidian.md`;
	});

	const input = `![[Obsidian]]

---

![[Obsidian#Link]]
`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Blockquotes', () => {
	const input = `> Human beings face ever more complex and urgent problems, and their effectiveness in dealing with these problems is a matter that is critical to the stability and continued progress of society.

- Doug Engelbart, 1961
`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Inline Code', () => {
	const input = `### Text inside \`backticks\` on a line will be formatted like code.
`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Code Blocks', () => {
	const input = `
\`\`\`dockerfile
FROM ubuntu
\`\`\`
`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Code Blocks with multiple $ characters', () => {
	const input = `
\`\`\`dockerfile
USER $USER_NAME:$USER_NAME
\`\`\`
`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Tables', () => {
	const input = `First Header | Second Header
------------ | ------------
Content from cell 1 | Content from cell 2
Content in the first column | Content in the second column
`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Footnotes', () => {
	const input = `Here's a simple footnote[^1]

[^1]: meaningful!

`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Footnotes in Tables', () => {
	const input = `| Table     | Col 1 | Col 2 | Col 3 |
	| --------- | ----- | ----- | ----- |
	| Footnotes | [^1]  | [^2]  | [^3]  | 
	
	[^1]: fn1
	[^2]: fn2
	[^3]: fn3
`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Math', () => {
	const input = `$$\begin{vmatrix}a & b\\
c & d
end{vmatrix}=ad-bc$$

You can also do inline math like $s^{-2}_{n}sum_{i=1}^{n}$`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Math with Code Blocks', () => {
	const input = `$$\begin{vmatrix}a & b\\
c & d
end{vmatrix}=ad-bc$$

\`\`\`dockerfile
USER $USER_NAME:$USER_NAME
\`\`\`

You can also do inline math like $s^{-2}_{n}sum_{i=1}^{n}$`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Math with Multiple Code Blocks', () => {
	const input = `$$\begin{vmatrix}a & b\\
c & d
end{vmatrix}=ad-bc$$

\`\`\`dockerfile
USER $USER_NAME:$USER_NAME
\`\`\`

You can also do inline math like $s^{-2}_{n}sum_{i=1}^{n}$

\`\`\`bash
eval "$(/home/$USER_NAME/.rbenv/bin/rbenv init -)"
\`\`\`

That was ruby, now we have javascript:

\`\`\`
console.log("Hello world!")
\`\`\`
`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Mermaid', () => {
	const input = `---
theme: beige
highlightTheme: css/vs2015.css

---

\`\`\`mermaid
sequenceDiagram
    Alice->>+John: Hello John, how are you?
    Alice->>+John: John, can you hear me?
    John-->>-Alice: Hi Alice, I can hear you!
    John-->>-Alice: I feel great!
\`\`\``;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Callouts', () => {
	const input = `> [!tip] This tip has a custom Header
	> This is an Tip with a custom Title
	<!-- element style="width:40%"-->
	
	> [!warning] 
	> This warning is aligned left
	<!-- element style="width:20%;font-size:24px" align="left"-->
	
	<br>
	
	> [!error] 
	> Callouts support manipulation through annotations
	<!-- element style="width:20%;font-size:24px" rotate="15"-->
	
	<grid drag="30 30" drop="11 68" style="font-size:36px">
	> [!example]
	> Here is an Example for an Callout in a Slide. Callouts support dark and white backgrounds and could be sized by annotations
	</grid>
	
	<grid drag="50 100" drop="50 0" bg="white" pad="50px" style="font-size:36px">
	> [!cite]
	> Here is an Cite for an Callout in a Slide. Callouts support dark and white backgrounds and could be sized by annotations
	</grid>`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});
