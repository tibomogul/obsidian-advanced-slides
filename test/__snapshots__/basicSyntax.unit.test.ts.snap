// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Markdown Syntax > Blockquotes 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

> Human beings face ever more complex and urgent problems, and their effectiveness in dealing with these problems is a matter that is critical to the stability and continued progress of society.

- Doug Engelbart, 1961
</div>"
`;

exports[`Basic Markdown Syntax > Callouts 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

<div class=\\"callout callout-color2\\">
<div class=\\"callout-title\\">
<div class=\\"callout-icon\\">

<i class=\\"fas fa-fire-alt\\" ></i>


</div>
<div class=\\"callout-title-inner\\">

This tip has a custom Header

</div>
</div>
<div class=\\"callout-content\\">

This is an Tip with a custom Title

</div>
</div>
	<!-- .element: style=\\"width: 40%\\" -->
	
<div class=\\"callout callout-color5\\">
<div class=\\"callout-title\\">
<div class=\\"callout-icon\\">

<i class=\\"fas fa-exclamation-triangle\\" ></i>


</div>
<div class=\\"callout-title-inner\\">

Warning

</div>
</div>
<div class=\\"callout-content\\">

This warning is aligned left

</div>
</div>
	<!-- .element: style=\\"width: 20%; font-size: 24px; display: flex; flex-direction: column; align-items: flex-start; justify-content: space-evenly\\" align=\\"left\\" -->
	
	<br>
	
<div class=\\"callout callout-color7\\">
<div class=\\"callout-title\\">
<div class=\\"callout-icon\\">

<i class=\\"fas fa-bolt\\" ></i>


</div>
<div class=\\"callout-title-inner\\">

Error

</div>
</div>
<div class=\\"callout-content\\">

Callouts support manipulation through annotations

</div>
</div>
	<!-- .element: style=\\"width: 20%; font-size: 24px; transform: rotate(15deg)\\" -->
	
	<div class=\\"\\" style=\\"font-size: 36px; position: absolute; left: 11%; top: 68%; height: 30%; width: 30%; display: flex; flex-direction: column; align-items: center; justify-content: center\\" >

<div class=\\"callout callout-color8\\">
<div class=\\"callout-title\\">
<div class=\\"callout-icon\\">

<i class=\\"fas fa-list\\" ></i>


</div>
<div class=\\"callout-title-inner\\">

Example

</div>
</div>
<div class=\\"callout-content\\">

Here is an Example for an Callout in a Slide. Callouts support dark and white backgrounds and could be sized by annotations

</div>
</div>
	</div>
	
	<div class=\\"has-light-background\\" style=\\"font-size: 36px; background-color: white; padding: 50px; box-sizing: border-box; position: absolute; left: 50%; top: 0%; height: 100%; width: 50%; display: flex; flex-direction: column; align-items: center; justify-content: center\\" >

<div class=\\"callout callout-color9\\">
<div class=\\"callout-title\\">
<div class=\\"callout-icon\\">

<i class=\\"fas fa-quote-left\\" ></i>


</div>
<div class=\\"callout-title-inner\\">

Cite

</div>
</div>
<div class=\\"callout-content\\">

Here is an Cite for an Callout in a Slide. Callouts support dark and white backgrounds and could be sized by annotations

</div>
</div>
	</div>
</div>"
`;

exports[`Basic Markdown Syntax > Code Blocks 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

\`\`\`dockerfile
FROM ubuntu
\`\`\`
</div>"
`;

exports[`Basic Markdown Syntax > Code Blocks with multiple $ characters 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

\`\`\`dockerfile
USER $USER_NAME:$USER_NAME
\`\`\`
</div>"
`;

exports[`Basic Markdown Syntax > Embeds 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

# Notes about Obsidian
A knowledge base that works on local Markdown files

# Link<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

Link to Obsidian Homepage: http://obsidian.md
</div></div>
---

Link to Obsidian Homepage: http://obsidian.md
"
`;

exports[`Basic Markdown Syntax > Footnotes 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

Here's a simple footnote<sup id=\\"fnref:1\\" role=\\"doc-noteref\\">1</sup>




<div class=\\"footnotes\\" role=\\"doc-endnotes\\">
<ol>
<li id=\\"fn:1\\" role=\\"doc-endnote\\" class=\\"footnote\\"><p>

meaningful!

</p></li></ol>
</div>
</div>"
`;

exports[`Basic Markdown Syntax > Footnotes in Tables 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

| Table     | Col 1 | Col 2 | Col 3 |
	| --------- | ----- | ----- | ----- |
| Footnotes |<sup id=\\"fnref:1\\" role=\\"doc-noteref\\">1</sup>|<sup id=\\"fnref:2\\" role=\\"doc-noteref\\">2</sup>|<sup id=\\"fnref:3\\" role=\\"doc-noteref\\">3</sup>|
	




<div class=\\"footnotes\\" role=\\"doc-endnotes\\">
<ol>
<li id=\\"fn:1\\" role=\\"doc-endnote\\" class=\\"footnote\\"><p>

fn1

</p></li><li id=\\"fn:2\\" role=\\"doc-endnote\\" class=\\"footnote\\"><p>

fn2

</p></li><li id=\\"fn:3\\" role=\\"doc-endnote\\" class=\\"footnote\\"><p>

fn3

</p></li></ol>
</div>
</div>"
`;

exports[`Basic Markdown Syntax > Headers 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

# This is a heading 1
## This is a heading 2
### This is a heading 3 
#### This is a heading 4
##### This is a heading 5
###### This is a heading 6
</div>"
`;

exports[`Basic Markdown Syntax > Images 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

Insert image with standard markdown syntax

<img src=\\"https://picsum.photos/id/1006/500/300\\" alt=\\"Image\\" style=\\"object-fit: scale-down\\">
</div>
---
<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

Insert image that lies outside of your vault

<img src=\\"/localFileSlash/Users/testUser/Desktop/howToUse.png\\" alt=\\"Image\\" style=\\"object-fit: scale-down\\">
</div>
---
<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

Insert image with obsidian markdown syntax

<img src=\\"/documentation/Image.jpg\\" alt=\\"\\" style=\\"object-fit: scale-down\\">
</div>
---
<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

Scale image to a width of 100 px

<img src=\\"/documentation/Image.jpg\\" alt=\\"\\" style=\\"width: 100px; object-fit: fill\\">
</div>
---
<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

Scale image to a width of 300x100 px

<img src=\\"/documentation/Image.jpg\\" alt=\\"\\" style=\\"object-fit: fill; width: 300px; height: 100px\\">
</div>"
`;

exports[`Basic Markdown Syntax > Inline Code 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

### Text inside \`backticks\` on a line will be formatted like code.
</div>"
`;

exports[`Basic Markdown Syntax > Links 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

External Links

http://obsidian.md - automatic!

[Obsidian](http://obsidian.md)
</div>
---
<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

Obsidian URI links

[Link to note](obsidian://open?path=D:%2Fpath%2Fto%2Ffile.md)

[Link to note](obsidian://open?vault=MainVault&file=MyNote.md)
</div>
---
<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

Internal Links

This Internal link will be displayed as normal text

This Link will use its alias for displaying
</div>"
`;

exports[`Basic Markdown Syntax > Links 2`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

Internal Links

This [Internal link](obsidian://open?vault=test-vault&file=Internal%20link) will be displayed as normal text

This [Link](obsidian://open?vault=test-vault&file=Internal%20link) will use its alias for displaying
</div>"
`;

exports[`Basic Markdown Syntax > Lists 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

- Item 1
- Item 2
	- Item 2a
	- Item 2b
</div>
---
<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

1. Item 1
1. Item 2
1. Item 3
   1. Item 3a
   1. Item 3b
</div>"
`;

exports[`Basic Markdown Syntax > Math 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

$egin{vmatrix}a & b\\\\
c & d
end{vmatrix}=ad-bc$

You can also do inline math like \`$s^{-2}_{n}sum_{i=1}^{n}$\`
</div>"
`;

exports[`Basic Markdown Syntax > Math with Code Blocks 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

$egin{vmatrix}a & b\\\\
c & d
end{vmatrix}=ad-bc$

\`\`\`dockerfile
USER $USER_NAME:$USER_NAME
\`\`\`

You can also do inline math like \`$s^{-2}_{n}sum_{i=1}^{n}$\`
</div>"
`;

exports[`Basic Markdown Syntax > Math with Multiple Code Blocks 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

$egin{vmatrix}a & b\\\\
c & d
end{vmatrix}=ad-bc$

\`\`\`dockerfile
USER $USER_NAME:$USER_NAME
\`\`\`

You can also do inline math like $s^{-2}_{n}sum_{i=1}^{n}$

\`\`\`bash
eval \\"$(/home/$USER_NAME/.rbenv/bin/rbenv init -)\\"
\`\`\`

That was ruby, now we have javascript:

\`\`\`
console.log(\\"Hello world!\\")
\`\`\`
</div>"
`;

exports[`Basic Markdown Syntax > Mermaid 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

<div class=\\"mermaid\\">
sequenceDiagram
    Alice->>+John: Hello John, how are you?
    Alice->>+John: John, can you hear me?
    John-->>-Alice: Hi Alice, I can hear you!
    John-->>-Alice: I feel great!

</div>
</div>"
`;

exports[`Basic Markdown Syntax > Tables 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

First Header | Second Header
------------ | ------------
Content from cell 1 | Content from cell 2
Content in the first column | Content in the second column
</div>"
`;

exports[`Basic Markdown Syntax > Text style 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

*This text will be italic*

_This will also be italic_

**This text will be bold**

**This will also be bold**



_You **can** combine them_
</div>
---
<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

Any word wrapped with two tildes (like ~~this~~) will appear crossed out.

Any word wrapped with two equal signs (like <mark>this</mark>) will appear as highlighted.
</div>"
`;
